File: dsa_leetcode_beginner/0252-meeting-rooms/0252-meeting-rooms.cpp

#include <iostream>
#include <vector>
#include <algorithm>

bool canAttendMeetings(std::vector<std::pair<int, int>> intervals) {
    std::sort(intervals.begin(), intervals.end());
    
    for (int i = 1; i < intervals.size(); i++) {
        if (intervals[i].first < intervals[i - 1].second) {
            return false;
        }
    }

    return true;
}

int main() {
    std::vector<std::pair<int, int>> intervals = {{0, 30}, {5, 10}, {15, 20}};
    std::cout << (canAttendMeetings(intervals) ? "true" : "false") << std::endl;
    return 0;
}

File: dsa_leetcode_beginner/0252-meeting-rooms/0252-meeting-rooms.py
def can_attend_meetings(intervals):
    intervals.sort()
    
    for i in range(1, len(intervals)):
        if intervals[i][0] < intervals[i - 1][1]:
            return False
        
    return True


#Question: https://www.lintcode.com/problem/920/
#Blog: https://blog.unwiredlearning.com/meeting-rooms